### DISCUSSION TITLE
After updating from 10.6 pimcore v to 11 version, facing version count issue

### DISCUSSION THREAD

**Discussion by Thejaswini-Rao-U** [pimcore/pimcore#16425] [Q&A / Support]
### Pimcore version

pimcore v 11.0

### Steps to reproduce

Hi

I have updated pimcore 10.6 v to pimcore 11, followed all the steps as per pimcore documentation. 
After login to admin, gives version count error.

### Actual Behavior

After login to admin UI, gives version count error.

Error attached here
![pimcore11](https://github.com/pimcore/pimcore/assets/78895325/f28c41f1-d6b6-475c-8d1d-7dc5bdf45206)

### Expected Behavior

It should work without error

Please let me know how to solve this?

thanks in advance

**Comment by Thejaswini-Rao-U** [comment]
Anyone please suggest?

**Comment by dpfaffenbauer** [comment]
https://pimcore.com/docs/platform/Pimcore/Installation_and_Upgrade/Updating_Pimcore/Preparing_for_V11#migrate-o_-prefix-properties-in-the-stored-data

**Comment by Thejaswini-Rao-U** [comment]
@dpfaffenbauer, Thanks for the response,

i executed these command 
**composer require --no-update pimcore/compatibility-bridge-v10** && 

**bin/console doctrine:migration:exec 'Pimcore\Bundle\CoreBundle\Migrations\Version20230508121105'**

followed same document before raising issue here.

but error still exists on admin panel after login
when im rebuilding classes command gives below error
i could not upload screenshot here so attaching error msg got in command line

**Saving all classes
11:55:03 ERROR     [pimcore] Class: Pimcore\Model\DataObject\FileExtensionMaster => call to undefined method seto_versionCount
11:55:03 EMERGENCY [pimcore] Exception: Call to undefined method seto_versionCount in class Pimcore\Model\DataObject\FileExtensionMaster in /var/www/html/projectName/vendor/pimcore/pimcore/lib/Model/AbstractModel.php:231
Stack trace:
#0 /var/www/html/projectName/vendor/pimcore/pimcore/lib/Model/AbstractModel.php(189): Pimcore\Model\AbstractModel->__call()
#1 /var/www/html/projectName/vendor/pimcore/pimcore/lib/Model/AbstractModel.php(170): Pimcore\Model\AbstractModel->setValue()
#2 /var/www/html/projectName/vendor/pimcore/pimcore/lib/Model/Dao/DaoTrait.php(39): Pimcore\Model\AbstractModel->setValues()
#3 /var/www/html/projectName/vendor/pimcore/pimcore/models/DataObject/Concrete/Dao.php(68): Pimcore\Model\Dao\AbstractDao->assignVariablesToModel()
#4 /var/www/html/projectName/vendor/pimcore/pimcore/models/DataObject/AbstractObject.php(244): Pimcore\Model\DataObject\Concrete\Dao->getById()
#5 /var/www/html/projectName/vendor/pimcore/pimcore/models/DataObject/Listing/Dao.php(66): Pimcore\Model\DataObject\AbstractObject::getById()
#6 [internal function]: Pimcore\Model\DataObject\Listing\Dao->load()
#7 /var/www/html/projectName/vendor/pimcore/pimcore/lib/Model/AbstractModel.php(220): call_user_func_array()
#8 /var/www/html/projectName/src/Service/OptionProviderFEM.php(52): Pimcore\Model\AbstractModel->__call()
#9 /var/www/html/projectName/src/Service/OptionProviderFEM.php(26): App\Service\OptionProviderFEM->getChannelOpt()
#10 /var/www/html/projectName/vendor/pimcore/pimcore/models/DataObject/ClassDefinition/Data/Multiselect.php(451): App\Service\OptionProviderFEM->getOptions()
#11 /var/www/html/projectName/vendor/pimcore/pimcore/models/DataObject/ClassDefinition.php(342): Pimcore\Model\DataObject\ClassDefinition\Data\Multiselect->preSave()
#12 /var/www/html/projectName/vendor/pimcore/pimcore/lib/Model/DataObject/ClassDefinition/ClassDefinitionManager.php(81): Pimcore\Model\DataObject\ClassDefinition->save()**


As i see in the Pimcore\Model\DataObject\FileExtensionMaster  class there is no method defined as seto_versionCount ? 
also i have searched whole project folder with this method  i didnt find any or similar to that?

the service file OptionProviderFEM.php pointing error, doesnt have any such method?
line no it showing has method  **$optionsData->setOrderKey("masterOrder");**

in the fileExtensionMaster class has method below which is pointing in the service file

/**
* Set masterOrder - Order
* @param float $masterOrder
* @return \Pimcore\Model\DataObject\FileExtensionMaster
*/
public function setMasterOrder ($masterOrder) {
	$fd = $this->getClass()->getFieldDefinition("masterOrder");
	$this->masterOrder = $fd->preSetData($this, $masterOrder);
	return $this;
}

regarding this gives error i dont have any idea of this?
how to solve this?

**Comment by Thejaswini-Rao-U** [comment]
Hi Team

Any response?

**Comment by dpfaffenbauer** [comment]
Did you actually read what I wrote? There is a separate script to migrate versions.

**Comment by Thejaswini-Rao-U** [comment]
@dpfaffenbauer , 

sorry i missed it, now i have read the scripts
but how to execute it? it should be only inside project root? not under any other folders?
let me know how to execute it?

**Comment by dpfaffenbauer** [comment]
> # Migrate o_ prefix properties in the stored data
> As o_ prefix will be removed from data objects system properties in v11. It is recommended to migrate the stored data to use new properties (without o_ prefix). Please adapt and use these [scripts](https://gist.github.com/dvesh3/50a1a99fd337d461e1652f2fd3b4d6cd) to migrate versions and recycle-bin data.

**Comment by Thejaswini-Rao-U** [comment]
@dpfaffenbauer , I got it, but how to execute it?
i have placed these 2 files inside project folder?
i tried using bin/console doctrine:migrations:migrate pimcore10.6_migrate_o_prefixes_recyclebin
didnt work
how to execute it please let me knw?

**Comment by dpfaffenbauer** [comment]
```php filename-of-script.php```

**Comment by Thejaswini-Rao-U** [comment]
@dpfaffenbauer , Thanks for response

i executed 2 commands. In the version command getting below error attached

command => php pimcore10.6_migrate_o_prefixes_versions.php

**Exception {#1022
  #message: "Call to undefined method seto_versionCount in class Pimcore\Model\DataObject\Tier"
  #code: 0
  #file: "./vendor/pimcore/pimcore/lib/Model/AbstractModel.php"
  #line: 231
  trace: {
    ./vendor/pimcore/pimcore/lib/Model/AbstractModel.php:231 { …}
    ./vendor/pimcore/pimcore/lib/Model/AbstractModel.php:189 { …}
    ./vendor/pimcore/pimcore/lib/Model/AbstractModel.php:170 { …}
    ./vendor/pimcore/pimcore/lib/Model/Dao/DaoTrait.php:39 { …}
    ./vendor/pimcore/pimcore/models/DataObject/Concrete/Dao.php:68 { …}
    ./vendor/pimcore/pimcore/models/DataObject/AbstractObject.php:244 { …}
    ./vendor/pimcore/pimcore/models/Element/AbstractElement.php:644 { …}
    ./vendor/pimcore/pimcore/models/DataObject/Concrete.php:736 { …}
    Pimcore\Model\DataObject\Concrete->__wakeup() {}
    ./vendor/pimcore/pimcore/lib/Tool/Serialize.php:31 { …}
    ./vendor/pimcore/pimcore/models/Version.php:290 { …}
    ./vendor/pimcore/pimcore/models/Version.php:413 { …}
    ./pimcore10.6_migrate_o_prefixes_versions.php:26 {
      › $data = null;
      › $data = $version->getData();
      › 
    }
  }
}**

**Comment by Thejaswini-Rao-U** [comment]
anyone please let me know how to solve this?

**Comment by Thejaswini-Rao-U** [comment]
Hi

Anyone please let me know...it will be very helpful?

**Reply by Fahl-Design** [reply]
@Thejaswini-Rao-U
you need to run this script with the lastest 10.6 version
than update to v11
if you have a lot of version files, this can take some days...

**Reply by Thejaswini-Rao-U** [reply]
@Fahl-Design , Thanks for the reply

I tried executing script in the 10.6 version, but shows below error

![migrate10Version](https://github.com/pimcore/pimcore/assets/78895325/c2fd5d29-5d8e-4395-89ae-e23f3ef31f19)

I tried giving permission for the file and owner but dint help

any idea?

**Reply by Fahl-Design** [reply]
maybe check the parent folder (`object/` ) permissions as well
make sure it's writable

**Reply by Thejaswini-Rao-U** [reply]
@Fahl-Design , thanks for the response
migration of version is complete.. Now i can move to version 11 to set up the project?
is this final migration should be executed on 10.6 v?
there is no error im getting, all the things are working fine on 10.6v so i can move to 11 version to setup?

**Comment by Thejaswini-Rao-U** [comment]
Hi Team,

I have executed script for versions in the 10.6 v => php pimcore10.6_migrate_o_prefixes_versions.php
started upgrading to 11 version again, now while executing migrate command  - **bin/console doctrine:migrations:migrate** , 
shows error attached below

**Are you sure you wish to continue? (yes/no) [yes]:
 > yes

[notice] Migrating up to Elements\Bundle\ProcessManagerBundle\Migrations\Version20230425000002
[error] Migration Pimcore\Bundle\CoreBundle\Migrations\Version20220614115124 failed during Execution. Error: "An exception occurred while executing a query: SQLSTATE[22032]: <<Unknown error>>: 3140 Invalid JSON text: "Invalid encoding in string." at position 247 in value for column '#sql-548_fad.data'."

In ExceptionConverter.php line 117:
                                                                                                                                             
  An exception occurred while executing a query: SQLSTATE[22032]: <<Unknown error>>: 3140 Invalid JSON text: "Invalid encoding in string."   
  at position 247 in value for column '#sql-548_fad.data'.                                                                                   
                                                                                                                                             

In Exception.php line 28:
                                                                                                                                             
  SQLSTATE[22032]: <<Unknown error>>: 3140 Invalid JSON text: "Invalid encoding in string." at position 247 in value for column '#sql-548_f  
  ad.data'.                                                                                                                                  
                                                                                                                                             

In Connection.php line 71:
                                                                                                                                             
  SQLSTATE[22032]: <<Unknown error>>: 3140 Invalid JSON text: "Invalid encoding in string." at position 247 in value for column '#sql-548_f  
  ad.data'.**                                                                                                                                  
                                                                                                


anyone has idea about this error?

**Reply by Thejaswini-Rao-U** [reply]
Anyone please let me know, will be helpful?

**Reply by Thejaswini-Rao-U** [reply]
i have solved above issue.. now im getting object id issue

This similar issue arrised before executing the script php pimcore10.6_migrate_o_prefixes_versions.php

now i have executed script and once again migrating from 10.6 to v 11, still this issue exists while executing **bin/console doctrine:migrations:migrate**

10:49:51 EMERGENCY [pimcore] PDOException: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'object_FEM.id' in 'field list' in /var/www/html/projectfolder/vendor/doctrine/dbal/src/Driver/PDO/Connection.php:71
Stack trace:
#0 /var/www/html/projectfolder//vendor/doctrine/dbal/src/Driver/PDO/Connection.php(71): PDO->query()
#1 /var/www/html/projectfolder/vendor/doctrine/dbal/src/Driver/Middleware/AbstractConnectionMiddleware.php(33): Doctrine\DBAL\Driver\PDO\Connection->query()
#2 /var/www/html/projectfolder//vendor/doctrine/dbal/src/Logging/Connection.php(43): Doctrine\DBAL\Driver\Middleware\AbstractConnectionMiddleware->query()
#3 /var/www/html/projectfolder//vendor/doctrine/dbal/src/Driver/Middleware/AbstractConnectionMiddleware.php(33): Doctrine\DBAL\Logging\Connection->query()
#4 /var/www/html/projectfolder//vendor/symfony/doctrine-bridge/Middleware/Debug/DBAL3/Connection.php(58): Doctrine\DBAL\Driver\Middleware\AbstractConnectionMiddleware->query()
#5 /var/www/html/projectfolder//vendor/doctrine/dbal/src/Connection.php(1100): Symfony\Bridge\Doctrine\Middleware\Debug\DBAL3\Connection->query()
#6 /var/www/html/projectfolder//vendor/doctrine/dbal/src/Connection.php(952): Doctrine\DBAL\Connection->executeQuery()
#7 /var/www/html/projectfolder//vendor/pimcore/pimcore/models/DataObject/Listing/Dao.php(105): Doctrine\DBAL\Connection->fetchFirstColumn()
#8 /var/www/html/projectfolder//vendor/pimcore/pimcore/models/DataObject/Listing/Concrete/Dao.php(45): Pimcore\Model\DataObject\Listing\Dao->loadIdList()
#9 /var/www/html/projectfolder//vendor/pimcore/pimcore/models/DataObject/Listing/Dao.php(62): Pimcore\Model\DataObject\Listing\Concrete\Dao->loadIdList()
#10 [internal function]: Pimcore\Model\DataObject\Listing\Dao->load()
#11 /var/www/html/projectfolder//vendor/pimcore/pimcore/lib/Model/AbstractModel.php(220): call_user_func_array()
#12 /var/www/html/projectfolder//src/Service/OptionProviderFEM.php(52): Pimcore\Model\AbstractModel->__call()
#13 /var/www/html/projectfolder//src/Service/OptionProviderFEM.php(26): App\Service\OptionProviderFEM->getChannelOpt()
#14 /var/www/html/projectfolder//vendor/pimcore/pimcore/models/DataObject/ClassDefinition/Data/Multiselect.php(451): App\Service\OptionProviderFEM->getOptions()
#15 /var/www/html/projectfolder//vendor/pimcore/pimcore/models/DataObject/ClassDefinition.php(342): Pimcore\Model\DataObject\ClassDefinition\Data\Multiselect->preSave()
#16 /var/www/html/projectfolder//vendor/pimcore/pimcore/bundles/CoreBundle/src/Migrations/Version20210107103923.php(40): Pimcore\Model\DataObject\ClassDefinition->save()
#17 /var/www/html/projectfolder//vendor/doctrine/migrations/lib/Doctrine/Migrations/Version/DbalExecutor.php(138): Pimcore\Bundle\CoreBundle\Migrations\Version20210107103923->up()
#18 /var/www/html/projectfolder/vendor/doctrine/migrations/lib/Doctrine/Migrations/Version/DbalExecutor.php(73): Doctrine\Migrations\Version\DbalExecutor->executeMigration()
#19 /var/www/html/projectfoldervendor/doctrine/migrations/lib/Doctrine/Migrations/DbalMigrator.php(88): Doctrine\Migrations\Version\DbalExecutor->execute()
#20 /var/www/html/projectfolder/vendor/doctrine/migrations/lib/Doctrine/Migrations/DbalMigrator.php(54): Doctrine\Migrations\DbalMigrator->executePlan()
#21 /var/www/html/projectfolder/vendor/doctrine/migrations/lib/Doctrine/Migrations/DbalMigrator.php(136): Doctrine\Migrations\DbalMigrator->executeMigrations()
#22 /var/www/html/projectfolder/vendor/doctrine/migrations/lib/Doctrine/Migrations/Tools/Console/Command/MigrateCommand.php(213): Doctrine\Migrations\DbalMigrator->migrate()
#23 /var/www/html/projectfolder/vendor/symfony/console/Command/Command.php(326): Doctrine\Migrations\Tools\Console\Command\MigrateCommand->execute()
#24 /var/www/html/projectfolder/vendor/symfony/console/Application.php(1096): Symfony\Component\Console\Command\Command->run()
#25 /var/www/html/projectfolder//vendor/symfony/framework-bundle/Console/Application.php(126): Symfony\Component\Console\Application->doRunCommand()
#26 /var/www/html/projectfolder//vendor/symfony/console/Application.php(324): Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand()
#27 /var/www/html/projectfolder//vendor/symfony/framework-bundle/Console/Application.php(80): Symfony\Component\Console\Application->doRun()
#28 /var/www/html/projectfolder//vendor/symfony/console/Application.php(175): Symfony\Bundle\FrameworkBundle\Console\Application->doRun()
#29 /var/www/html/projectfolder//vendor/symfony/runtime/Runner/Symfony/ConsoleApplicationRunner.php(49): Symfony\Component\Console\Application->run()
#30 /var/www/html/projectfolder/vendor/autoload_runtime.php(29): Symfony\Component\Runtime\Runner\Symfony\ConsoleApplicationRunner->run()


anyone please help me to solve this?

**Reply by Thejaswini-Rao-U** [reply]
anyone please let me know if you know about this error?

**Reply by Thejaswini-Rao-U** [reply]
anyone please let me know

**Reply by Thejaswini-Rao-U** [reply]
Hi 
please help me im getting below error while executing migration

error] Migration Pimcore\Bundle\CoreBundle\Migrations\Version20221222181745 failed during Execution. Error: "An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'object_classificationstore_groups_1.id' in 'where clause'"

In ExceptionConverter.php line 67:
                                                                                                                                             
  An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'object_classificationstore_groups_  
  1.id' in 'where clause'                                                                                                                    
                                                                                                                                             

In Exception.php line 28:
                                                                                                                     
  SQLSTATE[42S22]: Column not found: 1054 Unknown column 'object_classificationstore_groups_1.id' in 'where clause'  
                                                                                                                     

In Connection.php line 71:
                                                                                                                     
  SQLSTATE[42S22]: Column not found: 1054 Unknown column 'object_classificationstore_groups_1.id' in 'where clause'

**Comment by Thejaswini-Rao-U** [comment]
can anyone reply for my issue?

**Comment by Thejaswini-Rao-U** [comment]
Hi

on backend admin interface still getting this issue?

![versionCOuntJan23error](https://github.com/pimcore/pimcore/assets/78895325/c4c51ea7-f767-48c5-ac09-0a06a9b77bbd)
![versionCriticalErrorJan23](https://github.com/pimcore/pimcore/assets/78895325/43dcd17d-5779-4ad7-9b32-032dcabfa88c)
![versionCount23](https://github.com/pimcore/pimcore/assets/78895325/4e494318-54b5-4e57-aa13-98fe39024793)

Anyone please let me know how to solve this?

**Reply by Thejaswini-Rao-U** [reply]
i noticed, in the classes table, few of the columns are migrated, few are not, still objects table has o_versionCount, thats causes issue, how to migrate specific table columns to get it resolve issues?

few classification store, collection table has o prefixes, how to migrate it?

anyone?
